Backpack-SwiftUI/CardList/Classes/BPKCardList.swift

```diff
@@ -0,0 +1,134 @@
+        cardListSkeleton(with: sectionHeaderAction) {
+            ScrollView(.horizontal) {
+                HStack(spacing: .base) {
+                    ForEach(elements) { index in
+                        cardForElement(index)
+                    }
+                }
+                .padding(.horizontal, .base)
+            }
+        }
+    }
+
+    private var filteredCards: ArraySlice<Element> {
+        let cardsShown = (showingAllCards) ? elements.count : min(initiallyShownCardsCount, elements.count)
+        return elements[0..<cardsShown]
+    }
+
+    private func stackLayout(with accessory: BPKCardListLayout.Accessory?) -> some View {
+        cardListSkeleton(with: accessory?.sectionHeaderAction) {
+            ForEach(filteredCards) { element in
+                cardForElement(element)
+            }
+            .padding(.horizontal, .base)
+            accessoryView(accessory: accessory)
+                .padding(.horizontal, .base)
+        }
+    }
+
+    private func sectionHeader(with sectionHeaderAction: BPKCardListLayout.SectionHeaderAction?) -> some View {
+        if let sectionHeaderAction {
+            return BPKSectionHeader(title: title, description: description) {
+                BPKButton(
+                    icon: sectionHeaderAction.icon,
+                    accessibilityLabel: sectionHeaderAction.accessibilityLabel,
+                    action: sectionHeaderAction.action)
+                .buttonStyle((sectionHeaderAction.style == .default) ? .primary : .primaryOnDark)
+            }
+            .padding(.base)
+        } else {
+            return BPKSectionHeader(title: title, description: description)
+                .padding(.base)
+        }
+    }
+
+    private func cardListSkeleton<LayoutContent: View>(
+        with sectionHeaderAction: BPKCardListLayout.SectionHeaderAction?,
+        @ViewBuilder andContent content: () -> LayoutContent) -> some View {
+            VStack(alignment: .leading, spacing: .base) {
+                sectionHeader(with: sectionHeaderAction)
+
+                if elements.count > 0 {
+                    content()
+                }
+            }
+        }
+
+    @ViewBuilder private func accessoryView(accessory: BPKCardListLayout.Accessory?) -> some View {
+        if case .footerButton(let button) = accessory {
+            BPKButton(
+                button.title,
+                icon: button.icon,
+                action: button.action
+            )
+            .buttonStyle((button.style == .default) ? .primary : .primaryOnDark)
+            .stretchable()
+        }
+        if case .expand(let expandingText, let collapsingText, _) = accessory {
+            BPKButton(
+                (showingAllCards) ? collapsingText : expandingText
+            ) {
+                withAnimation {
+                    showingAllCards.toggle()
+                }
+            }
+            .buttonStyle(.link)
+            .stretchable()
+        }
+    }
```

Comment:
Should we make an extension on the Accessory enum to provide a `var accessory`, this way we can switch over all cases and have all layouts neatly organised. 

as for calling `showingAllCards.toggle()` we can resolve that with a button.action closure

Reply:
> Should we make an extension on the Accessory enum to provide a var accessory, this way we can switch over all cases and have all layouts neatly organised.

Not sure I understood you correctly, `Accessory` enum is already the accessory. We could add a computed variable to `Layout` to provide an accessory if it applies, but not sure how it would improve.
Using `if case` instead of `switch` here we decided with @frugoman to include section header button as an extra accessory option, and using `switch` forced us to provide `EmptyView` as it's content.

> as for calling showingAllCards.toggle() we can resolve that with a button.action closure
Can add a closure to configure it, but what I understood is that the expand accessory option provides a built in functionality that's not as configurable